### ISSUE TITLE
Module build failing -  PostCSS plugin post-css-flexbugs-fixes requires PostCSS 8

### ISSUE THREAD

**Issue by charterbran** [gatsbyjs/gatsby#29933]
## Module build failing - PostCSS plugin post-css-flexbugs-fixes requires PostCSS 8

Running gatsby dev yields about 100 webpack errors, all related to my CSS files. Localhost:8000 displays a blank page. This is after I updated all of my dependencies by running npm outdated and installing each one @latest, as well as updating to Gatsby 3.0.0. following the migration guide. 

Errors all look like this, but for each CSS module file I have:

```shell
ERROR in ./src/templates/tagsTemplate.css
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
ModuleBuildError: Module build failed (from ./node_modules/postcss-loader/dist/cjs.js):
Error: PostCSS plugin postcss-flexbugs-fixes requires PostCSS 8.
Migration guide for end-users:
https://github.com/postcss/postcss/wiki/PostCSS-8-for-end-users
    at Processor.normalize (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/postcss/lib/processor.es6:117:15)
    at new Processor (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/postcss/lib/processor.es6:38:25)
    at postcss (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/postcss/lib/postcss.es6:34:10)
    at Object.loader (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/postcss-loader/dist/index.js:87:20)
    at processResult (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/webpack/lib/NormalModule.js:598:19)
    at /Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/webpack/lib/NormalModule.js:692:5
    at /Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/loader-runner/lib/LoaderRunner.js:399:11
    at /Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/loader-runner/lib/LoaderRunner.js:251:18
    at context.callback (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/loader-runner/lib/LoaderRunner.js:124:13)
    at Object.loader (/Users/branoneusebio/Documents/Software Dev/charter-official/node_modules/postcss-loader/dist/index.js:96:7)
 @ ./src/templates/tagsTemplate.js 8:0-28
 @ ./.cache/_this_is_virtual_fs_path_/$virtual/async-requires.js 171:11-173:5
 @ ./.cache/app.js 17:0-52 28:0-70 30:27-40 28:0-70
 ```

### Steps to reproduce

Running gatsby develop causes the error. It's a production site and the repo is private but if needed I can try to make a new repo with same code as a reproducible copy. 

### Expected result

Running gatsby develop should build the site with no ESlint errors or anything.

### Actual result

Running gatsby develop should throw about 91 webpack errors as seen above, with this message at the end: 
91 ERRORS in child compilations (Use 'stats.children: true' resp. '--stats-children' for more details)
webpack compiled with 182 errors

Additionally, localhost:8000 will be a blank page vs. the entire site. 

### Environment

```
  System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.18.2 - /usr/local/bin/node
    npm: 7.6.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.192
    Safari: 13.1
  npmPackages:
    gatsby: ^3.0.0 => 3.0.0
    gatsby-image: ^3.0.0 => 3.0.0
    gatsby-plugin-google-analytics: ^3.0.0 => 3.0.0
    gatsby-plugin-google-gtag: ^3.0.0 => 3.0.0
    gatsby-plugin-manifest: ^3.0.0 => 3.0.0
    gatsby-plugin-netlify: ^3.0.0 => 3.0.0
    gatsby-plugin-offline: ^4.0.0 => 4.0.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.0.0
    gatsby-plugin-sharp: ^3.0.0 => 3.0.0
    gatsby-plugin-webfonts: ^1.1.4 => 1.1.4
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-transformer-remark: ^3.0.0 => 3.0.0
    gatsby-transformer-sharp: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    gatsby-cli: 3.0.0
```

**Comment by vladar** [comment]
Some community plugins may be not compatible yet (or maybe you are using packages that depend on older versions of webpack/postcss, etc).

Please use `yarn why post-css-flexbugs-fixes` (or `npm ls post-css-flexbugs-fixes`) to see the chain of dependencies. I suspect you have two versions of webpack.

Maybe you will have to stick with yarn resolutions for some time (until all other packages use the latest deps). See this section of the migration guide: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#handling-dependencies-for-plugins-that-are-not-yet-updated

**Comment by charterbran** [comment]
@vladar Thanks for your response and for your help.

I ran that command and it came back with: 
gatsby-starter-default@0.1.0 /Users/branoneusebio/Documents/Software Dev/charter-official
└── (empty)

What exactly does that mean? And if I have two versions of Webpack how can I mitigate that? 

Also - is this not going to be a problem for everyone else? I haven't done any custom configuration to the webpack config or to css modules, etc. I'm using what came out of the box essentially as far as I know.

**Comment by charterbran** [comment]
Running npm update gives this message:

npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: gatsby-starter-default@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.0.0" from google-map-react@2.1.9
npm ERR! node_modules/google-map-react
npm ERR!   google-map-react@"^2.1.9" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/branoneusebio/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/branoneusebio/.npm/_logs/2021-03-02T21_38_23_810Z-debug.log

**Comment by charterbran** [comment]
I got the site running by adding the postcss dependencies into my package.json and forcing the update from an older version to 8+. I don't fully understand what was happening but gatsby develop now builds the site per usual without those warnings from webpack. 

{
  "name": "gatsby-starter-default",
  "private": true,
  "description": "A simple starter to get up and developing quickly with Gatsby",
  "version": "0.1.0",
  "author": "Kyle Mathews <mathews.kyle@gmail.com>",
  "dependencies": {
    "aos": "^2.3.4",
    "chart.js": "^2.9.4",
    "gatsby": "^3.0.0",
    "gatsby-image": "^3.0.0",
    "gatsby-plugin-google-analytics": "^3.0.0",
    "gatsby-plugin-google-gtag": "^3.0.0",
    "gatsby-plugin-manifest": "^3.0.0",
    "gatsby-plugin-netlify": "^3.0.0",
    "gatsby-plugin-offline": "^4.0.0",
    "gatsby-plugin-react-helmet": "^4.0.0",
    "gatsby-plugin-sharp": "^3.0.0",
    "gatsby-plugin-webfonts": "^1.1.4",
    "gatsby-source-filesystem": "^3.0.0",
    "gatsby-transformer-remark": "^3.0.0",
    "gatsby-transformer-sharp": "^3.0.0",
    "google-map-react": "^2.1.9",
    **"postcss": "^8.1.7",
    "postcss-flexbugs-fixes": "^5.0.2",
    "postcss-loader": "^5.0.0",**
    "prop-types": "^15.7.2",
    "react": "^17.0.1",
    "react-chartjs-2": "^2.11.1",
    "react-dom": "^17.0.1",
    "react-google-recaptcha": "^2.1.0",
    "react-helmet": "^6.1.0",
    "react-share": "^4.4.0",
    "react-simple-chatbot": "^0.6.1",
    "styled-components": "^5.2.1"
  },
  "devDependencies": {
    "prettier": "^2.2.1"
  },
  "keywords": [
    "gatsby"
  ],
  "license": "0BSD",
  "scripts": {
    "build": "gatsby build",
    "develop": "gatsby develop",
    "format": "prettier --write \"**/*.{js,jsx,ts,tsx,json,md}\"",
    "start": "npm run develop",
    "serve": "gatsby serve",
    "clean": "gatsby clean",
    "test": "echo \"Write tests! -> https://gatsby.dev/unit-testing\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/gatsbyjs/gatsby-starter-default"
  },
  "bugs": {
    "url": "https://github.com/gatsbyjs/gatsby/issues"
  }
}

**Comment by vladar** [comment]
Hey @charterbran 

Can you provide [minimal reproduction][reproduction] of the original issue? We can double-check what exactly was wrong there and maybe document in the migration guide.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by AbhyudayaSharma** [comment]
@vladar I'm also facing the same issue. Although this is not really minimal, I am seeing this issue in this repository: https://github.com/AbhyudayaSharma/abhyudayasharma.github.io/tree/gatsby-3

Corresponding GH actions failure logs: https://pipelines.actions.githubusercontent.com/pC5rfCyB5cNASJ8N3Z6Z0ybtglNsEG4dL1kEVsPSg7JPfvb6bt/_apis/pipelines/1/runs/1282/signedlogcontent/3?urlExpires=2021-03-03T10%3A57%3A31.0646717Z&urlSigningMethod=HMACV1&urlSignature=47JPJFW%2B8E7jJ1wB2sr5ECdc00%2Bfxp0Hz6eeEMxIPMw%3D

**Comment by vladar** [comment]
@AbhyudayaSharma several dependencies in your project still require postcss7:

- `stylelint` (and its subdependencies)
- `gatsby-transformer-remark` -> `sanitize-html`
- `gatsby` -> `css-minimizer-webpack-plugin` -> `cssnano`
- `gatsby-plugin-sass` - >`resolve-url-loader`

I will double-check those in Gatsby dependency chain - we will upgrade those as soon as possible. In the meantime - you will likely have to add `postcss 8` explicitly to your package.json OR use `yarn resolutions` and pin `postcss` to `8.2.6` ([see this suggestion in the migration guide](https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#handling-dependencies-for-plugins-that-are-not-yet-updated)).

**Comment by johngrasty** [comment]
For what it's worth, I had the same problem. I blew away my `node_modules` directory and switched from `npm` to `yarn`, and everything is fine now.

**Comment by thomasjungblut** [comment]
As @vladar mentioned, I just got it fixed by putting a resolution for postcss into the package.json:

```
  "resolutions": {
    "graphql": "^15.4.0",
    "graphql-compose": "^7.25.0",
    "webpack": "^5.24.2",
    "postcss": "8.2.6"
  },
```

Then install `npm i postcss-flexbugs-fixes`, which installs 5.0.2 at the minute. Seems to work fine so far, but I'm not done testing everything.

**Comment by AbhyudayaSharma** [comment]
For NPM, installing `postcss@latest` fixed the issue. Thanks a lot @vladar !

**Comment by vladar** [comment]
Thank you for opening this, @AbhyudayaSharma 

We're marking this issue as answered and closing it but please feel free to comment and mention me if you would like to continue this discussion. We hope we managed to help and thank you for using Gatsby! 💜

**Comment by leesgithub2019** [comment]
> Some community plugins may be not compatible yet (or maybe you are using packages that depend on older versions of webpack/postcss, etc).
> 
> Please use `yarn why post-css-flexbugs-fixes` (or `npm ls post-css-flexbugs-fixes`) to see the chain of dependencies. I suspect you have two versions of webpack.
> 
> Maybe you will have to stick with yarn resolutions for some time (until all other packages use the latest deps). See this section of the migration guide: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#handling-dependencies-for-plugins-that-are-not-yet-updated



I checked my dependency through `npm ls post-css-flexbugs-fixes` 
the results are as follows:
>
`
├─┬ @storybook/addon-essentials@6.2.9
│ └─┬ @storybook/addon-docs@6.2.9
│   └─┬ @storybook/builder-webpack4@6.2.9
│     └── postcss-flexbugs-fixes@4.2.1 
└── postcss-flexbugs-fixes@5.0.2 
`
There are two `postcss-flexbugs-fixes`
I deleted postcss-flexbugs-fixes from my package.json 
>
 then deleted node_modules ,
>
`npm install`
>
everything's fine !!!
